sentry-android-core/src/main/java/io/sentry/android/core/gestures/SentryGestureListener.java

```diff
@@ -0,0 +1,238 @@
+    }
+
+    final String direction = scrollState.calculateDirection(motionEvent);
+    addBreadcrumb(scrollTarget, scrollState.type,
+      Collections.singletonMap("direction", direction));
+    scrollState.reset();
+  }
+
+  @Override
+  public boolean onDown(final @Nullable MotionEvent motionEvent) {
+    if (motionEvent == null) {
+      return false;
+    }
+    scrollState.reset();
+    scrollState.startX = motionEvent.getX();
+    scrollState.startY = motionEvent.getY();
+    return false;
+  }
+
+  @Override
+  public boolean onSingleTapUp(final @Nullable MotionEvent motionEvent) {
+    final View decorView = ensureWindowDecorView("onSingleTapUp");
+    if (decorView == null || motionEvent == null) {
+      return false;
+    }
+
+    @SuppressWarnings("Convert2MethodRef")
+    final @Nullable View target =
+      ViewUtils.findTarget(decorView, motionEvent.getX(), motionEvent.getY(),
+        view -> ViewUtils.isViewTappable(view));
+
+    if (target == null) {
+      options
+        .getLogger()
+        .log(SentryLevel.INFO, "Unable to find click target. No breadcrumb captured.");
+      return false;
+    }
+
+    addBreadcrumb(target, "click");
+    return false;
+  }
+
+  @Override
+  public boolean onScroll(
+    final @Nullable MotionEvent firstEvent,
+    final @Nullable MotionEvent currentEvent,
+    final float distX,
+    final float distY
+  ) {
+    final View decorView = ensureWindowDecorView("onScroll");
+    if (decorView == null || firstEvent == null) {
+      return false;
+    }
+
+    if (scrollState.type == null) {
+      @SuppressWarnings("Convert2MethodRef")
+      final @Nullable View target =
+        ViewUtils.findTarget(decorView, firstEvent.getX(), firstEvent.getY(),
+          view -> ViewUtils.isViewScrollable(view));
+
+      if (target == null) {
+        options
+          .getLogger()
+          .log(SentryLevel.INFO, "Unable to find scroll target. No breadcrumb captured.");
+        return false;
+      }
+
+      scrollState.setTarget(target);
+      scrollState.type = "scroll";
+    }
+    return false;
+  }
+
+  @Override
+  public boolean onFling(
+    final @Nullable MotionEvent motionEvent,
+    final @Nullable MotionEvent motionEvent1,
+    final float v,
+    final float v1
+  ) {
+    scrollState.type = "swipe";
+    return false;
+  }
+
+  @Override
+  public void onShowPress(MotionEvent motionEvent) {
+  }
+
+  @Override
+  public void onLongPress(MotionEvent motionEvent) {
+  }
+
+  // region utils
+  private void addBreadcrumb(
+    final @NotNull View target,
+    final @NotNull String eventType
+  ) {
+    addBreadcrumb(target, eventType, Collections.emptyMap());
+  }
+
+  private void addBreadcrumb(
+    final @NotNull View target,
+    final @NotNull String eventType,
+    final @NotNull Map<String, Object> additionalData
+  ) {
+    @NotNull String className;
+    if (target.getClass().getCanonicalName() != null) {
+      className = target.getClass().getCanonicalName();
```

Comment:
Is this likely going to be minified? If so, we might need to consider proguard support here with the native team

Reply:
Yes, it will be minified for some cases (like AndroidX views), but this is the same problem we have now with the fragment integration, so I guess they should be tackled together

Reply:
The other integrations use `simpleName`, should we keep `getCanonicalName`? Also, I'd cache the `getCanonicalName` in a local field since it does some work and we're calling twice.

Reply:
I think canonical name makes sense here, what if you have a view that you named the same, but it's your custom one (like `io.sentry.Button`)? simple name wouldn't give you a clue what button is this.